questionpy_server/worker/__init__.py

```diff
@@ -188,3 +190,11 @@ async def get_static_file(self, path: str) -> PackageFileData:
     @abstractmethod
     async def get_static_file_index(self) -> dict[str, PackageFile]:
         """Returns the index of static files as declared in the package's manifest."""
+
+    def get_loaded_packages(self, *, only_with_hash: bool = True) -> list[LoadedPackage]:
+        """Get all loaded packages in the worker.
+
+        Args:
+            only_with_hash: Only return packages that have a hash (i.e. only ZIP packages).
+        """
+        return [p for p in self.loaded_packages if p.hash is not None or not only_with_hash]
```

Comment:
Würde ich zwecks Konsistenz auch in `abstractmethod` hier und Implementierung in `BaseWorker` trennen. Aus `Worker` könnte man eigentlich auch mal ein `Protocol` machen

Reply:
Stimmt, das sollte getrennt sein.